bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/discovery/DigiplexDiscoveryService.java

```diff
@@ -137,21 +131,25 @@ public void handleAreaLabelResponse(@NonNull AreaLabelResponse response) {
 
     @Override
     public void setThingHandler(@Nullable ThingHandler handler) {
-        bridgeHandler = (DigiplexBridgeHandler) handler;
-        if (bridgeHandler != null) {
+        if (handler instanceof DigiplexBridgeHandler) {
+            bridgeHandler = (DigiplexBridgeHandler) handler;
             bridgeHandler.registerMessageHandler(this);
         }
     }
 
     @Override
-    @Nullable
-    public ThingHandler getThingHandler() {
+    public @Nullable ThingHandler getThingHandler() {
         return bridgeHandler;
     }
 
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
     @Override
     public void deactivate() {
+        removeOlderResults(getTimestampOfLastScan());
```

Comment:
Do we want this to happen?

Reply:
No, we do not. I can remember. But there are tons of bindings doing it at the time the service will be deactivated. I will change it for the touched bindings in this PR.